deeplab/README.md

```diff
@@ -0,0 +1,202 @@
+import {getLabels, getColormap} from '@tensorflow-models/deeplab';
+const model = 'ade20k';
+const colormap = getColormap(model);
+const labels = getLabels(model);
+```
+
+### Segmenting an Image
+
+The `segment` method of the `SemanticSegmentation` object covers most use cases.
+
+Each model recognises a different set of object classes in an image:
+
+- [PASCAL](./deeplab/src/config.ts#L60)
+- [CityScapes](./deeplab/src/config.ts#L66)
+- [ADE20K](./deeplab/src/config.ts#L72)
+
+#### `model.segment` input
+
+- **image** :: `ImageData | HTMLImageElement | HTMLCanvasElement | HTMLVideoElement | tf.Tensor3D`;
+
+The image to segment
+
+- **canvas** (optional) :: `HTMLCanvasElement`
+
+Pass an optional canvas element as `canvas` to draw the output as a side effect.
+
+- **colormap** (optional) :: `[number, number, number][]`
+
+The array of RGB colors corresponding to labels
+
+- **labels** (optional) :: `string[]`
+
+The array of names corresponding to labels
+
+By [default](./src/index.ts#L81), `colormap` and `labels` are set according to the `base` model attribute passed during initialization.
+
+#### `model.segment` output
+
+The output is a promise of a `DeepLabOutput` object, with four attributes:
+
+- **legend** :: `{ [name: string]: [number, number, number] }`
+
+The legend is a dictionary of objects recognized in the image and their colors in RGB format.
+
+- **height** :: `number`
+
+The height of the returned segmentation map
+
+- **width** :: `number`
+
+The width of the returned segmentation map
+
+- **segmentationMap** :: `Uint8ClampedArray`
+
+The colored segmentation map as `Uint8ClampedArray` which can be [fed](https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Tutorial/Pixel_manipulation_with_canvas) into `ImageData` and mapped to a canvas.
+
+#### `model.segment` example
+
+```typescript
+const classify = async (image) => {
+    return await model.segment(image);
+}
+```
+
+**Note**: *For more granular control, consider `predict` and `toSegmentationImage` methods described below.*
+
+### Producing a Semantic Segmentation Map
+
+To segment an arbitrary image and generate a two-dimensional tensor with class labels assigned to each cell of the grid overlayed on the image (with the maximum number of cells on the side fixed to 513), use the `predict` method of the `SemanticSegmentation` object.
+
+#### `model.predict` input
+
+- **image** :: `ImageData | HTMLImageElement | HTMLCanvasElement | HTMLVideoElement | tf.Tensor3D`;
+
+The image to segment
+
+#### `model.predict` output
+
+- **rawSegmentationMap** :: `Promise<tf.Tensor2D>`
+
+The segmentation map of the image
+
+#### `model.predict` example
+
+```javascript
+const getSemanticSegmentationMap = async (image) => {
+    return await model.predict(image)
+}
+```
+
+### Translating a Segmentation Map into the Color-Labelled Image
+
+To transform the segmentation map into a coloured image, use the `toSegmentationImage` method of the `SemanticSegmentation` object.
```

Comment:
`SemanticSegmentation` does not have a `toSegmentationImage` method and are part of `utils`.